File: src/test/java/de/zalando/zally/rules/SuccessResponseAsJsonObjectRuleTest.java
package de.zalando.zally.rules;

import io.swagger.models.Swagger;
import io.swagger.parser.SwaggerParser;
import org.junit.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

public class SuccessResponseAsJsonObjectRuleTest {

    @Test
    public void responseAsJsonObjectRuleNoViolationsJson() throws IOException {
        SuccessResponseAsJsonObjectRule rule = new SuccessResponseAsJsonObjectRule();
        Swagger swagger = new SwaggerParser().read("/api_spp.json");
        assertThat(rule.validate(swagger)).isEmpty();
    }

    @Test
    public void responseAsJsonObjectRuleNoViolationYaml() throws IOException {
        SuccessResponseAsJsonObjectRule rule = new SuccessResponseAsJsonObjectRule();
        Swagger swagger = new SwaggerParser().read("/api_spa.yaml");
        assertThat(rule.validate(swagger).size()).isEqualTo(2);
    }
}

File: server/src/test/java/de/zalando/zally/rules/SuccessResponseAsJsonObjectRuleTest.kt
package de.zalando.zally.rules

import de.zalando.zally.getFixture
import org.assertj.core.api.Assertions.assertThat
import org.junit.Test

class SuccessResponseAsJsonObjectRuleTest {

    private val validSwagger = getFixture("successResponseAsJsonObjectValid.json")
    private val invalidSwagger = getFixture("successResponseAsJsonObjectInvalid.json")
    private val npeSwagger = getFixture("sample_swagger_api.yaml")

    @Test
    fun positiveCase() {
        assertThat(SuccessResponseAsJsonObjectRule().validate(validSwagger)).isNull()
    }

    @Test
    fun negativeCase() {
        val result = SuccessResponseAsJsonObjectRule().validate(invalidSwagger)!!
        assertThat(result.paths).hasSameElementsAs(listOf("/pets GET 200", "/pets POST 200"))
    }

    @Test
    fun positiveCaseSpp() {
        val swagger = getFixture("api_spp.json")
        assertThat(SuccessResponseAsJsonObjectRule().validate(swagger)).isNull()
    }

    @Test
    fun npeBug() {
        assertThat(SuccessResponseAsJsonObjectRule().validate(npeSwagger)).isNotNull()
    }
}
